src/firmware/app/primitives/primitive_manager.c

```diff
@@ -118,6 +110,9 @@ void app_primitive_manager_startNewPrimitive(PrimitiveManager_t *manager,
                                              FirmwareWorld_t *world,
                                              TbotsProto_Primitive primitive_msg)
 {
+    static_assert(TbotsProto_Primitive_estop_tag == 1,
+                  "E-Stop primitive not assigned tag of 1");
```

Comment:
Maybe add a short reason as to why this is a `static_assert` to prevent future members from removing this accidentally

Reply:
I actually don't understand this myself? The tag only really seems to matter in the switch statement for choosing a primitive to run, and since we're matching on the tag (variable) we shouldn't care what the actual value is?

Reply:
We don't want bad primitive proto msgs to cause the robots to _not_ e-stop. So we want to be super sure that if someone were to go change the primitive proto msg, the assumption about the estop primitive being first still holds

Reply:
I'm assuming by "bad" you don't mean malformed / corrupted right? You mean more a mistake in the proto msg definition / code? Being extra-safe with changing code around the estop and safety behaviour makes sense, but I don't understand whyt he estop primitive needs to be first in the primitive proto msg? There's no meaning to the order that I'm aware of and it shouldn't affect the conditional logic that selects the primitive either? Just clarifying here.

Seems the only way to "break" estop behaviour would be to _remove_ the e-stop primitive (either from the switch statement, or entirely), or to change the estop primitive logic in firmware. In either case I don't actually see how the `static_assert` prevents these changes on its own, since the code would still compile (I think).

Reply:
Yeah I meant proto msg definition re-ordering, not corrupt protos

If we updated the primitive proto and reshuffled the order of the primitives in the "one-of", the e-stop will no longer be the first tag, since all the primitive tags would get re-assigned accordingly. 

If we accidentally flash the robots with a different version of the primitive proto, with the `static_assert` being okay with the change, we can at least be sure that the e-stop will work, because its still the first tag and hasn't been re-assigned.

That being said, we would need a corresponding `static_assert` somewhere on the C++ side to make sure we aren't sending primitive protos that move the position of the e-stop primitive. The `static_assert` would _have_ to exist on both sides for this to make sense.

Reply:
Ah that makes more sense, I didn't think of mismatched / different firmware versions potentially messing things up since the tags/enums are really just ints.

If we did flash some robots with a different version of the primitive proto though, wouldn't it fail to decode entirely? I'm not sure if just re-ordering items in a `oneof` would be a "recoverable" error or not.

Reply:
It might not fail to decode, just because there's no data integrity "add-ons" like a hash built into the proto.

Reply:
we need to figure out a system to check hashes, but in the meantime, I think this static assert is helpful

Reply:
Yeah, agreed. I also agree with Mat that we should make sure to add a comment here explaining _why_ this is needed.

Reply:
I've added the comment